maildir/maildiropen.c

```diff
@@ -122,17 +123,30 @@ int maildir_safeopen_stat(const char *path, int mode, int perm,
 			, perm);
 
 	if (fd < 0)	return (fd);
-	if (fcntl(fd, F_SETFL, (mode & O_APPEND)) || fstat(fd, stat1)
-	    || lstat(path, &stat2))
+	if (fcntl(fd, F_SETFL, (mode & O_APPEND)) || stat1 && fstat(fd, stat1))
 	{
 		close(fd);
 		return (-1);
 	}
 
-	if (stat1->st_dev != stat2.st_dev || stat1->st_ino != stat2.st_ino)
+	p = getenv("MAILDIR_SKIP_SYMLINK_CHECKS");
+	if (p && atoi(p)) return (fd);
+
+	if (!stat1)
+	{
+		stat1 = &statt;
+		if (fstat(fd, stat1))
```

Comment:
I don't see that this is needed, at this point. We know that nothing needs stat() any more, at this point, so just proclaim success.

Reply:
No.  This is the old behaviour, ie, courier-imapd is NOT configured to skip the stat (symlink) check.

The original if at line 126 will only execute fstat() now if stat1 was non-NULL upon entry.

If MAILDIR_SKIP_SYMLINK_CHECKS is set and not 0 then by this point we'd already have returned (only performing fstat if the caller requires it), so by this point we are looking to perform then symlink check and if we have not already done so we need to issue fstat() so that we can proceed with the symlink check which now happens on line 146 below.